docs/storage.md

```diff
@@ -484,6 +485,68 @@ config:
 prefix: ""
 ```
 
+### OCI - Oracle Cloud Infrastructure
+
+To configure OCI Object Storage as Thanos Object Store, you need access to OCI Object Storage.
+OCI has 3 different providers; default, instance principal, and raw, which are supported by this object storage client implementation
+
+#### Default Provider
+The default config provider will look for configurations in 3 places: file in `$HOME/.oci/config`, `$HOME/.obmcs/config`, and
+variables names starting with the string TF_VAR. If the same configuration is found in multiple places the provider will
+prefer the first one.
+
+For Example:
+```yaml
+type: OCI
+config:
+  provider: "default"
+  bucket: ""
+  compartment_ocid: ""
+  part_size: ""                   // Optional part size to override the OCI default of 128 MiB, value is in bytes.
+  max_request_retries: ""         // Optional maximum number of retries for a request.
+  request_retry_interval: ""      // Optional sleep duration in seconds between retry requests.
+  http_config:
+    idle_conn_timeout: 1m30s      // Optional maximum amount of time an idle (keep-alive) connection will remain idle before closing itself. Zero means no limit.
+    response_header_timeout: 2m   // Optional amount of time to wait for a server's response headers after fully writing the request.
+    tls_handshake_timeout: 10s    // Optional maximum amount of time waiting to wait for a TLS handshake. Zero means no timeout.
+    expect_continue_timeout: 1s   // Optional amount of time to wait for a server's first response headers. Zero means no timeout and causes the body to be sent immediately.
+    insecure_skip_verify: false   // Optional. If true, crypto/tls accepts any certificate presented by the server and any host name in that certificate.
+    max_idle_conns: 100           // Optional maximum number of idle (keep-alive) connections across all hosts. Zero means no limit.
+    max_idle_conns_per_host: 100  // Optional maximum idle (keep-alive) connections to keep per-host. If zero, DefaultMaxIdleConnsPerHost=2 is used.
+    max_conns_per_host: 0         // Optional maximum total number of connections per host.
+    disable_compression: false    // Optional. If true, prevents the Transport from requesting compression.
+    client_timeout: 90s           // Optional time limit for requests made by the HTTP Client.
+```
+
+
```

Comment:
```suggestion

Alternatively, you can provide API signing key authentication parameters in the configuration:

```yaml
type: OCI
config:
  provider: "raw"
  bucket: "thanos" // name of the object storage bucket
  compartment_ocid: "ocid1.compartment.oc1..."  // compartment in which the object storage bucket was created
  tenancy_ocid: "ocid1.tenancy.oc1..."
  user_ocid: "ocid1.user.oc1..."
  region: "us-ashburn-1"
  fingerprint: "aa:a1:bb:b1:cc:c1:dd:d1:ee:e1:ff:f1:aa:a2:bb:b2"
  privatekey: "/path/to/oci_api_key.pem"
  passphrase: ""         // Optional passphrase used to encrypt the private API Signing key

```